src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -508,6 +508,7 @@ internal static void ReportReservedTypeName(string? name, CSharpCompilation comp
             if (reportIfContextual(SyntaxKind.RecordKeyword, MessageID.IDS_FeatureRecords, ErrorCode.WRN_RecordNamedDisallowed)
                 || reportIfContextual(SyntaxKind.RequiredKeyword, MessageID.IDS_FeatureRequiredMembers, ErrorCode.ERR_RequiredNameDisallowed)
                 || reportIfContextual(SyntaxKind.FileKeyword, MessageID.IDS_FeatureFileTypes, ErrorCode.ERR_FileTypeNameDisallowed)
+                || reportIfContextual(SyntaxKind.ClosedKeyword, MessageID.IDS_FeatureFileTypes, ErrorCode.ERR_ClosedTypeNameDisallowed)
```

Comment:
Do we have a test for this code path?
```suggestion
                || reportIfContextual(SyntaxKind.ClosedKeyword, MessageID.IDS_FeatureClosedClass, ErrorCode.ERR_ClosedTypeNameDisallowed)
```